src/mds/MDCache.cc

```diff
@@ -1418,6 +1423,49 @@ void MDCache::adjust_subtree_after_rename(CInode *diri, CDir *olddir, bool pop)
   show_subtrees();
 }
 
+// ===================================
+// Default cache memory limit
+
+
+/*
+ * Get the default cache memory limit using the mds_memory_target
+ */
+uint64_t MDCache::get_default_mds_cache_memory_limit()
+{
+  // RSS usage has been experimentally shown to be around 25 percent more than the cache_memory_limit. Hence, the cache_memory_limit is reset as per the mds_memory_target(which is an upper bound on the rss usage)
+  uint64_t temp_cache_limit = (uint64_t) (mds->memory_target/1.25);
+  mds->cct->_conf.set_val_default("mds_cache_memory_limit", stringify(temp_cache_limit));
```

Comment:
Let's move this method's logic to the constructor: Following that, initialize:

`cache_memory_limit = g_conf().get_val<Option::size_t>("mds_cache_memory_limit");`

Otherwise you are always setting `mds_cache_memory_limit` based on the default value of `memory_target` (or whatever it is set to).

It's also strange to mutate any state in a `get_*` method. Following these changes, we should be able to drop this method.